src/Drivers/PhpAmqpLibAmqpDriver.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Smuuf\CeleryForPhp\Drivers;
+
+use Smuuf\CeleryForPhp\StrictObject;
+use PhpAmqpLib\Wire\AMQPTable;
+use PhpAmqpLib\Message\AMQPMessage;
+use PhpAmqpLib\Exception\AMQPProtocolChannelException;
+use PhpAmqpLib\Exception\AMQPTimeoutException;
+
+use Smuuf\CeleryForPhp\Drivers\IAmqpDriver;
+
+/**
+ * AMQP driver backed by php-amqplib
+ */
+class PhpAmqpLibAmqpDriver implements IAmqpDriver {
+
+    use StrictObject;
+    
+    private $message = null;
+    private $channel = null;
+
+    public function __construct(
+        private $amqp,
+    ) {}
+
+    public function publish(string $queue, string $exchange, string $routingKey, string $message, array $properties, array $headers): bool {
+        $ch = $this->amqp->channel();
+        if(! empty($routingKey) ){
+            $ch->exchange_declare($exchange, 'direct', false, true, false);
+            $ch->queue_declare($queue, false, true, false, false);
+            $ch->queue_bind($queue, $exchange, $routingKey);
+        }else {
+            $ch->exchange_declare($exchange, 'fanout', false, true, false);
+        }
+        $properties['application_headers'] = new AMQPTable($headers);
+        $amqpMessage = new AMQPMessage($message, $properties);
+        $ch->basic_publish($amqpMessage, $exchange, $routingKey);
+        $ch->close();
+
+        return true;
+    }
+
+    public function deleteExchange(string $exchange){
+        $ch = $this->amqp->channel();
+
+        try {
+            $ch->exchange_delete($exchange);
+        } catch (\PhpAmqpLib\Exception\AMQPProtocolChannelException $e) {
+            if ($e->getCode() !== 404) {  // ignore exchange 404
+                throw $e;
+            }
+        }
+        $ch->close();
+
+        return true;
+    }
+
+    public function deleteQueue(string $queueName){
+        $ch = $this->amqp->channel();
+
+        try {
+            $ch->queue_delete($queueName);
+        } catch (\PhpAmqpLib\Exception\AMQPProtocolChannelException $e) {
+            if ($e->getCode() !== 404) {  // ignore queue 404
+                throw $e;
+            }
+        }
+        $ch->close();
+
+        return true;
+    }
+
+    public function purgeQueue(string $queueName){
+        $ch = $this->amqp->channel();
+
+        try {
+            $ch->queue_purge($queueName);
+        } catch (\PhpAmqpLib\Exception\AMQPProtocolChannelException $e) {
+            if ($e->getCode() !== 404) {  // ignore queue 404
+                throw $e;
+            }
+        }
+        $ch->close();
+
+        return true;
+    }
+
+    public function queueLength(string $queueName): int {
+        $ch = $this->amqp->channel();
+
+        list($queue, $messageCount) = $ch->queue_declare($queueName, false, true, false, false);
+
+        $ch->close();
+
+        return $messageCount;
+    }
+
+    public function fetchResult(string $task_id, int $expire=0, bool $removeFromQueue = true)
```

Comment:
This method doesn't seem to be used anywhere...? It seems like this is meant to be used in situations where AMQP/RabbitMQ should act as a result backend - but that's doesn't seem to be relevant for this PR _(and, again, this method is unused)_.

Consider removing this method _(from both here and from the `IAmqpDriver` interface, along with the `consumeResult` method called from this one)_.